### PR TITLE
Cleanup redis

### PR DESCRIPTION
**Pull request by bcmendoza**
https://app.shortcut.com/grey-matter/story/9507/sanitize-redis-connection-string-generated-by-operator
https://app.shortcut.com/grey-matter/story/9511/hide-redis-from-the-dashboard

In addition to addressing the issues linked, this PR also:
- Defaults Redis to an internally-managed cache. Those requiring an external Redis server should stick to helm charts.
- Removes Redis as a dependency for the JWT security service.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,19 @@
 
 ### Added
 
-- Add SPIRE as SPIFFE implementation for mutual TLS between workloads
+- SPIRE as SPIFFE implementation for mutual TLS between workloads
```
Comment (dontlaugh) [inline CHANGELOG.md:7]: Do these need to be under a vNEXT changelog section?

**Review by dontlaugh — CHANGES_REQUESTED** [review]
one nitpick

File: CHANGELOG.md (hunk 1 above)
Comment (bcmendoza) [reply CHANGELOG.md:7]: I actually haven't released 0.3.0 yet... would you like to me to update the heading so its vNEXT until I'm ready to cut an actual release? I figured we want to add some more changes before cutting 0.3.0.

Comment (dontlaugh) [reply CHANGELOG.md:7]: Yes, update it to vNEXT

**Review by dontlaugh — APPROVED** [review]

**Pull request merged**